aws/biganimal-csp-preflight

```diff
@@ -452,11 +481,17 @@ nlb_quota=$(aws service-quotas get-service-quota --region "$region" --service-co
 
 # calculate required resources
 need_mgmt_vcpus=$(infra_vcpus)
-need_pg_vcpus=$(need_pg_vcpus_for "$pg_vm_vcpu" "$ha")
-need_regional_vcpus=$((need_mgmt_vcpus + need_pg_vcpus))
+need_pg_vcpus=$(need_pg_vcpus_for "$pg_vm_vcpu" "$ha" "$architecture")
+need_pg_eha_proxy_vcpus=$(need_pg_eha_proxy_vcpus_for "$pg_eha_proxy_vm_vcpu" "$architecture")
+need_regional_vcpus=$((need_mgmt_vcpus + need_pg_vcpus + need_pg_eha_proxy_vcpus))
 need_eip=$(need_public_ip)
 need_vpc=$(need_vpc)
 need_nlb=1
+shared_mgmt_eha_proxy_vcpus=0
+
+if [ "$mgmt_vm_type" = "$ehaproxy_vm_type" ]; then
+    shared_mgmt_eha_proxy_vcpus=$((need_mgmt_vcpus + need_pg_eha_proxy_vcpus))
```

Comment:
To not display two times the suggestion of increasing the quota of `mgmt_vm_type` and `ehaproxy_vm_type` if they are exactly the same instance type (as it happens now that we are not allowing the final user to customize those, even if theoretically in the future we can support that), I've added also `shared_mgmt_eha_proxy_vcpus` to use that later (it's the sum of the needed vCPUs for both `mgmt_vm_type` and `ehaproxy_vm_type`). Then later we use this to check if we need to write in the suggestions table both the rows of `mgmt_vm_type` and `ehaproxy_vm_type` or if we need to write only the first row (because these are using exactly the same kind of instance type) increasing the number of vCPUs needed for this kind of instance type (with the sum of both `need_mgmt_vcpus` and `need_pg_eha_proxy_vcpus`). Feel free to suggest a better approach if you have something in mind! 👍

Reply:
Thanks! I think merging `need_mgmt_vcpus ` and `need_pg_eha_proxy_vcpus ` is quite reasonable when the type is same.